harness/tests/experiment/pytorch_utils.py

```diff
@@ -0,0 +1,265 @@
+# type: ignore
+import os
+import pathlib
+import random
+import sys
+import typing
+import uuid
+
+from torch.distributed import launcher
+
+import determined as det
+from determined import gpu, pytorch
+from tests.experiment import utils
+
+
+def calculate_gradients(
+    batch_size: int = 4, epoch_size: int = 64, num_epochs: int = 3
+) -> typing.List[float]:
+    # independently compute expected metrics
+    batches = [
+        (v[:], v[:])
+        for v in (
+            [x * 0.1 + 1.0 for x in range(y, y + batch_size)]
+            for y in (z % epoch_size for z in range(0, epoch_size * num_epochs, batch_size))
+        )
+    ]
+
+    lr = 0.001

```

Comment:
seems odd to hardcode this, maybe make it an optional parameter at least?

Reply:
Moved to function def, but moved it back to the test file since this is pretty specific to that one test.